### PR TITLE
Bug/#1043 popup notclosing clickoutside

### PR DESCRIPTION
**Pull request by desperado1802**
Fixed modals not closing on click outside or when pressed "Escape" key. Also added "X" icon that closes the modal. on top right

https://github.com/ever-co/ever-teams/assets/124465103/4346b13f-450c-4ac9-93d7-c99dcdf45275

### CODE REVIEW

**Comment by evereq** [comment]
@desperado1802 not good padding from border of popup, check some examples in the component libs

File: apps/web/lib/features/team/invite/invite-form-modal.tsx [hunk 1]
```diff
@@ -91,7 +91,7 @@ export function InviteFormModal({
 	);
 
 	return (
-		<Modal isOpen={open} closeModal={() => undefined}>
+		<Modal isOpen={open} closeModal={() => closeModal()}>
```
Comment (badalkhatri0924) [inline apps/web/lib/features/team/invite/invite-form-modal.tsx:94]: @desperado1802 

It should be `<Modal isOpen={open} closeModal={closeModal}>`

File: apps/web/lib/components/modal.tsx [hunk 2]
```diff
@@ -40,12 +41,22 @@ export function Modal({
...
 					className="absolute inset-0 flex items-center justify-center p-4"
 				>
-					<Dialog.Panel className={className}>
+					<Dialog.Panel className={`${className} relative`}>
 						{title && <Dialog.Title>{title}</Dialog.Title>}
 						{description && (
 							<Dialog.Description>{description}</Dialog.Description>
 						)}
-
+						<div
```
Comment (badalkhatri0924) [inline apps/web/lib/components/modal.tsx:49]: @desperado1802 Keep checking responsive versions too.

image

**Review by badalkhatri0924 — CHANGES_REQUESTED** [review]

File: apps/web/lib/components/modal.tsx (hunk 2 above)
Comment (desperado1802) [reply apps/web/lib/components/modal.tsx:49]: Improved the position and respobsivness, also called the close function directly
image
image

**Comment by evereq** [comment]
@badalkhatri0924 / @desperado1802 - let me know when it's possible to merge and also please fix merge conflict before that. Thanks!

File: apps/web/lib/components/modal.tsx (hunk 2 above)
Comment (badalkhatri0924) [reply apps/web/lib/components/modal.tsx:49]: > Improved the position and respobsivness, also called the close function directly image
> 
> image

@desperado1802 Top and Right padding should be same, not looking good in responsive device.

Comment (evereq) [reply apps/web/lib/components/modal.tsx:49]: image

"Top and Right padding should be same" 100% agree with @badalkhatri0924 . In fact, it's one of the very important "features" (if not most important) of great engineers - attention to detail...

**Comment by evereq** [comment]
@desperado1802 are we using features of https://headlessui.com/react/dialog#basic-example and https://headlessui.com/react/dialog#showing-and-hiding-your-dialog ? I.e. " clicking outside the Dialog.Panel or pressing the Escape key will fire the close event and close the dialog." that is a built-in feature, did you use it!?

**Comment by desperado1802** [comment]
> @desperado1802 are we using features of https://headlessui.com/react/dialog#basic-example and https://headlessui.com/react/dialog#showing-and-hiding-your-dialog ? I.e. " clicking outside the Dialog.Panel or pressing the Escape key will fire the close event and close the dialog." that is a built-in feature, did you use it!?

Yes @evereq  we are using those features of the headless ui. The modal is closing when pressed outside of the container or when pressed Escape key.


https://github.com/ever-co/ever-teams/assets/124465103/077e2b20-8b10-4105-bd47-3848815f9beb

**Review by badalkhatri0924 — APPROVED** [review]

**Comment by badalkhatri0924** [comment]
@evereq we can merge this PR.

**Pull request merged**